### PR TITLE
Quota plugin configurable

### PR DESCRIPTION
**Pull request by at88mph**
Some code cleanup of unused code as well.

### CODE REVIEW

File: cavern/README.md [hunk 1]
```diff
@@ -57,6 +57,12 @@ org.opencadc.cavern.resourceID = ivo://{authority}/{name}
 # (optional) identify which container nodes are allocations
 org.opencadc.cavern.allocationParent = {top level node}
 
+# (optional) provide a class implementing the org.opencadc.cavern.nodes.QuotaPlugin interface to control Quotas
+# for users.
+# Optional, but the default of NoQuotaPlugin will get used if not specified, and users will have free reign.
+#
```
Comment (pdowler) [inline cavern/README.md:63]: need to document the choices in the text below

In fenwick and tantar we had plugins that are internal (artifactSelector and ResolutionPolicy respectively) which we started out putting fully qualified class names and then changed to a simple set of terms and mapped that to classes internally. I think for internally implemented (which I expect the quota plugins will be) that's slightly nicer and it gets rid of really dynamic plugin loading.

I'd probably follow the style in tantar.

File: cavern/src/main/java/org/opencadc/cavern/CavernConfig.java [hunk 2]
```diff
@@ -220,7 +221,68 @@ public Subject getRootOwner() {
...
+     *
+     * @param <T>             Class type of the instantiated class
+     * @param implementationClassName   Class name to create
+     * @param constructorArgs The constructor arguments
+     * @return configured implementation of the interface
+     *
+     * @throws IllegalStateException if an instance cannot be created
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T loadPlugin(final String implementationClassName, final Object... constructorArgs)
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/CavernConfig.java:256]: might not need this if using a hard coded name -> impl mapping

File: cavern/src/main/java/org/opencadc/cavern/nodes/FileSystemNodePersistence.java [hunk 3]
```diff
@@ -183,15 +182,16 @@ public FileSystemNodePersistence() {
...
             try {
                 URL baseURL = posixMapperID.toURL();
-                this.posixMapper = new MyPosixMapperClient(baseURL);
+                posixMapper = new MyPosixMapperClient(baseURL);
             } catch (MalformedURLException ex) {
                 throw new InvalidConfigException("invalid " + Standards.POSIX_GROUPMAP.toASCIIString() + " base URL: " + posixMapperID, ex);
             }
         } else {
-            this.posixMapper = new MyPosixMapperClient(posixMapperID);
+            posixMapper = new MyPosixMapperClient(posixMapperID);
```
Comment (pdowler) [inline cavern/src/main/java/org/opencadc/cavern/nodes/FileSystemNodePersistence.java:194]: prefer to keep the `this.` prefix on the left hand side when assigning to member vars; it stands out when a bit better in a non-syntax highlighting environment

**Review by pdowler — CHANGES_REQUESTED** [review]

File: cavern/src/main/java/org/opencadc/cavern/nodes/FileSystemNodePersistence.java (hunk 3 above)
Comment (at88mph) [reply cavern/src/main/java/org/opencadc/cavern/nodes/FileSystemNodePersistence.java:194]: The `posixMapper` variable was moved into a local variable as it was only used in creating the `GroupCache` object.  It wasn't used as a Class member.

Comment (pdowler) [reply cavern/src/main/java/org/opencadc/cavern/nodes/FileSystemNodePersistence.java:194]: ah, ok. makes sense.

**Review by pdowler — APPROVED** [review]

**Pull request merged**